static/app/components/searchSyntax/utils.tsx

```diff
@@ -54,3 +54,31 @@ export function treeTransformer(
 
   return tree.map(nodeVisitor);
 }
+
+type GetKeyNameOpts = {
+  aggregateWithArgs?: boolean;
+};
+
+/**
+ * Utility to get the string name of any type of key.
+ */
+export const getKeyName = (
+  key: ReturnType<
+    TokenConverter['tokenKeySimple' | 'tokenKeyExplicitTag' | 'tokenKeyAggregate']
+  >,
+  options?: GetKeyNameOpts
+) => {
+  const {aggregateWithArgs} = options ?? {};
```

Comment:
You can have a default in the args too, which might be nicer

```tsx
export const getKeyName = (
  key: ReturnType<
    TokenConverter['tokenKeySimple' | 'tokenKeyExplicitTag' | 'tokenKeyAggregate']
  >,
  options?: GetKeyNameOpts = {}
) => {
```

Reply:
oooh that is nicer